lib/msf/core/post/windows/version.rb

```diff
@@ -0,0 +1,150 @@
+# -*- coding: binary -*-
+
+module Msf::Post::Windows::Version
+include Msf::Post::Windows::Registry
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+            ]
+          }
+        }
+      )
+    )
+  end
+
+  def get_version_info
+    result = get_version_info_impl
+    if result.nil?
+      print_error("Couldn't retrieve the target's build number!")
+      raise RuntimeError.new("Couldn't retrieve the target's build number!")
+    end
+
+    result
+  end
+
+  def get_version_info_fallback_impl
+        build_num_raw = cmd_exec('ver')
+        groups = build_num_raw.match(/.*Version\s+(\d+)\.(\d+)\.(\d+)(\.(\d+))?/)
+        if groups.nil?
+          return nil
+        end
+
+        major, minor, build, unused, revision = groups.captures
+        revision = 0 if revision.nil?
```

Comment:
It looks like `revision` is not used anywhere. I might be missing something though.

Reply:
Ah yes, there are a few places throughout the rest of the codebase where Revision _is_ actually used. I intend to integrate it into the `WindowsVersion` code, and then use it in those specific locations, but this PR is getting big enough that I will circle back around and just do that. For now I'll remove it.